gocontext.go

```diff
@@ -64,13 +64,15 @@ func StartSpanOptions(ctx context.Context, name, spanType string, opts SpanOptio
 // set either from err, or from the caller.
 //
 // If there is no span or transaction in the context, CaptureError returns
-// nil. As a convenience, if the provided error is nil, then CaptureError
-// will also return nil.
+// Error with nil ErrorData field. As a convenience, if the provided error is nil,
```

Comment:
Can you please explain the need for this change? It's not clear to me, and I'm reluctant to change this behaviour.

Reply:
Wanted to avoid situation when returned type is nil pointer to interface - it can be tricky sometimes https://stackoverflow.com/questions/13476349/check-for-nil-and-nil-interface-in-go

Second thing is to keep original error even if error was already sent.

If you like, I can revert that change - was made to provide stable interface, no matter error was sent previously or not.

Reply:
I think I can see reason in that: we would be saying that if the result is nil if and only if the input error is nil. If the input error is non-nil, but there is no span or transaction in the context, then we'll get a non-nil Error that wraps the input error, only its (Send, SetTransaction, SetSpan) methods will be no-ops.

I'm fine with keeping this, but I'd like a change to the doc comment: it should be stated in terms of functionality, rather than whether the ErrorData field is nil or not. Maybe something like this?

> If the provided error is nil, then CaptureError will also return nil; otherwise a non-nil Error will always be returned. If there is no transaction or span in the context, then the returned Error's Send method will have no effect.